pkg/kubectl/resource_printer_test.go

```diff
@@ -276,12 +284,17 @@ func TestUnknownTypePrinting(t *testing.T) {
 
 func TestTemplateEmitsVersionedObjects(t *testing.T) {
 	// kind is always blank in memory and set on the wire
-	printer, err := NewTemplatePrinter([]byte(`{{.kind}}`), testapi.Version(), api.Scheme)
+	printer, err := NewTemplatePrinter([]byte(`{{.kind}}`))
```

Comment:
In general, I find our code dangerously under-commented. It's difficult to tell the intent and/or assumptions unless one understands all of the code, which few of us do.